### ISSUE TITLE
Support X-Forwarded-For header for start auth flow request

### ISSUE THREAD

**Issue by dmytro-bekuzarov** [TrueLayer/truelayer-java#194]
**Describe the bug**
Support X-Forwarded-For header for start auth flow request
https://docs.truelayer.com/reference/start-payment-authorization-flow

**To Reproduce**
1. Create a payment with the bank that requires user IP address for auth flow and start auth flow
2. See 400 Bad request error

**Expected behavior**
TrueLayer Java Client show allow to pass X-Forwarded-For header for start auth flow request

**Comment by dmytro-bekuzarov** [comment]
@dili91 Could you take a look at this issue? We are currently blocked by this for some banks.

**Comment by dili91** [comment]
Hey @dmytro-bekuzarov, thanks for raising this. This has now been implemented in version [6.1.0](https://github.com/TrueLayer/truelayer-java/releases/tag/6.1.0). 
I'm closing this for now, let us know if any issues! Thanks

**Comment by dili91** [comment]
Hey @dmytro-bekuzarov I'm reopening this issue to let you know that we're changing this in the next major version of the library (`8.0.0`). the details are here, but if you want to upgrade, the TLDR is that you will have to switch from this: 
```java
ApiResponse<AuthorizationFlowResponse> response = tlClient.payments()
                .startAuthorizationFlow(A_PAYMENT_ID, request, endUserIpAddress)
                .get();
```
to this: 
```
var headers = Headers.builder().xForwardedFor(endUserIpAddress).build();
ApiResponse<AuthorizationFlowResponse> response = tlClient.payments()
                .startAuthorizationFlow(headers, A_PAYMENT_ID, request)
                .get();
```

I know it's a bit unfortunate but as we had to anyway introduce a major release we preferred including now this breaking change.
